python/rikai/internal/reflection.py

```diff
@@ -22,3 +22,41 @@ def find_class(class_name: str):
     module, cls = class_name.rsplit(".", 1)
     mod = importlib.import_module(module)
     return getattr(mod, cls)
+
+
+def check_class(class_name: str):
+    """
+    Assuming `x.y.z.name` as class_name, check
+    if `from x.y.z import name` works
+    """
+    try:
+        module, cls = class_name.rsplit(".", 1)
+        mod = importlib.import_module(module)
+        return hasattr(mod, cls)
+    except (ModuleNotFoundError, ValueError):
+        return False
+
+
+def find_func(func_name: str):
+    module, cls, func = func_name.rsplit(".", 2)
+    try:
+        mod = importlib.import_module(module)
```

Comment:
how come this doesn't need to catch ModuleNotFoundError?

Reply:
Because to `find_func`, we must `check_func` first